### ISSUE TITLE
Replication of Jigsaw Toxicity Results

### ISSUE THREAD

**Issue by pruksmhc** [ghzhang233/Non-Discrimination-Learning-for-Text-Classification#10]
I am running the code (without any changes), and am not able to replicate the reported results. 
For the baseline model (without weights), I run `python main.py`, and get the below results. 
Unnamed: 0   acc_dev   auc_dev  acc_test  auc_test   acc_eec   auc_eec  efpr_eec  efnr_eec  eauc_eec
0           0  0.948390  0.926629  0.946831  0.924152  0.945013  0.994845  1.530650  0.312191  0.735914
1           1  0.947032  0.926879  0.946942  0.924213  0.942767  0.995171  1.368950  0.714149  0.935193
2           2  0.946942  0.928191  0.946257  0.925692  0.907359  0.980726  1.520012  3.543188  2.167654
3           3  0.947247  0.927384  0.946437  0.924081  0.975328  0.998819  0.995368  0.238841  0.521595
4           4  0.947143  0.926533  0.946298  0.923519  0.944190  0.993905  1.421833  1.408007  1.140447
5           5  0.946755  0.925790  0.946319  0.923033  0.946411  0.991710  0.733161  2.226244  1.317227
6           6  0.947960  0.926707  0.947012  0.923332  0.930228  0.984109  0.884181  4.361955  2.362223
7           7  0.946845  0.926851  0.945903  0.924031  0.954757  0.997575  1.276729  0.296264  0.736440
8           8  0.946104  0.926993  0.945806  0.924173  0.954443  0.997020  1.011461  0.324356  0.546204
9           9  0.946935  0.927221  0.945993  0.923317  0.971279  0.997793  1.233777  0.579832  0.770537

For the model (with instance weighting), I run `python main.py --use_weights --name_model debiased`, and get 
Unnamed: 0   acc_dev   auc_dev  acc_test  auc_test   acc_eec   auc_eec  efpr_eec  efnr_eec  eauc_eec
0           0  0.948390  0.926629  0.946831  0.924152  0.945013  0.994845  1.530650  0.312191  0.735914
1           1  0.947032  0.926879  0.946942  0.924213  0.942767  0.995171  1.368950  0.714149  0.935193
2           2  0.946942  0.928191  0.946257  0.925692  0.907359  0.980726  1.520012  3.543188  2.167654
3           3  0.947247  0.927384  0.946437  0.924081  0.975328  0.998819  0.995368  0.238841  0.521595
4           4  0.947143  0.926533  0.946298  0.923519  0.944190  0.993905  1.421833  1.408007  1.140447
5           5  0.946755  0.925790  0.946319  0.923033  0.946411  0.991710  0.733161  2.226244  1.317227
6           6  0.947960  0.926707  0.947012  0.923332  0.930228  0.984109  0.884181  4.361955  2.362223
7           7  0.946845  0.926851  0.945903  0.924031  0.954757  0.997575  1.276729  0.296264  0.736440
8           8  0.946104  0.926993  0.945806  0.924173  0.954443  0.997020  1.011461  0.324356  0.546204
9           9  0.946935  0.927221  0.945993  0.923317  0.971279  0.997793  1.233777  0.579832  0.770537

The mean for EFPR and EFNR for the baseline and instance weighting is pretty similar, with EFPR and EFNR for the baseline model being slightly lower than the model with instance weighting. 

Is there something I'm missing, or some hyperparameter set I should be tuning for?
Thank you!

**Comment by pruksmhc** [comment]
Ah, upon rereading the README, I see that I should be trying with seed 0 - will try that and reopen if i'm still having a replication issue!

**Comment by pruksmhc** [comment]
I'm still getting replication issues, even using the seed. 
Running the bsaeline code, I get the below:
            acc_dev    auc_dev   acc_test   auc_test    acc_eec    auc_eec   efpr_eec   efnr_eec   eauc_eec
count 10.000000  10.000000  10.000000  10.000000  10.000000  10.000000  10.000000  10.000000  10.000000
mean  0.941833   0.919084   0.942738   0.921996   0.943946   0.991490   0.946352   1.628285   1.072813
std      0.003356   0.001476   0.003694   0.001323   0.019085   0.005198   0.399517   0.914916   0.411891
min     0.936298   0.916860   0.936201   0.919865   0.917442   0.982796   0.427023   0.423484   0.616845
25%     0.939691   0.918310   0.941114   0.921200   0.932850   0.987383   0.829629   0.896174   0.747794
50%     0.943538   0.918872   0.944345   0.922198   0.939613   0.992211   0.900189   1.588936   1.022235
75%     0.944227   0.920286   0.945635   0.922841   0.958763   0.995717   0.933675   2.255161   1.276882
max     0.944989   0.921355   0.945793   0.923734   0.973891   0.997633   1.708225   3.302407   1.814090

With debiasing, numbers seem to get worse:
              acc_dev    auc_dev   acc_test   auc_test    acc_eec    auc_eec   efpr_eec   efnr_eec   eauc_eec
count  10.000000  10.000000  10.000000  10.000000  10.000000  10.000000  10.000000  10.000000  10.000000
mean    0.943308   0.918551   0.944318   0.921334   0.926287   0.987116   1.165896   2.200072   1.423375
std     0.000776   0.001829   0.000867   0.001990   0.012230   0.005744   0.425408   1.236855   0.399914
min     0.941907   0.915953   0.942524   0.917727   0.911551   0.973537   0.212848   0.436604   0.925831
25%     0.942884   0.917561   0.943938   0.920590   0.918686   0.986615   0.977410   1.535752   1.177314
50%     0.943434   0.918699   0.944373   0.921697   0.923495   0.987332   1.194725   1.884363   1.337604
75%     0.943630   0.918997   0.944830   0.922080   0.929108   0.989980   1.509363   3.033730   1.697343
max     0.944795   0.921647   0.945751   0.924552   0.951165   0.995651   1.631650   4.361005   2.192520

**Comment by ghzhang233** [comment]
> I am running the code (without any changes), and am not able to replicate the reported results.
> For the baseline model (without weights), I run `python main.py`, and get the below results.
> Unnamed: 0 acc_dev auc_dev acc_test auc_test acc_eec auc_eec efpr_eec efnr_eec eauc_eec
> 0 0 0.948390 0.926629 0.946831 0.924152 0.945013 0.994845 1.530650 0.312191 0.735914
> 1 1 0.947032 0.926879 0.946942 0.924213 0.942767 0.995171 1.368950 0.714149 0.935193
> 2 2 0.946942 0.928191 0.946257 0.925692 0.907359 0.980726 1.520012 3.543188 2.167654
> 3 3 0.947247 0.927384 0.946437 0.924081 0.975328 0.998819 0.995368 0.238841 0.521595
> 4 4 0.947143 0.926533 0.946298 0.923519 0.944190 0.993905 1.421833 1.408007 1.140447
> 5 5 0.946755 0.925790 0.946319 0.923033 0.946411 0.991710 0.733161 2.226244 1.317227
> 6 6 0.947960 0.926707 0.947012 0.923332 0.930228 0.984109 0.884181 4.361955 2.362223
> 7 7 0.946845 0.926851 0.945903 0.924031 0.954757 0.997575 1.276729 0.296264 0.736440
> 8 8 0.946104 0.926993 0.945806 0.924173 0.954443 0.997020 1.011461 0.324356 0.546204
> 9 9 0.946935 0.927221 0.945993 0.923317 0.971279 0.997793 1.233777 0.579832 0.770537
> 
> For the model (with instance weighting), I run `python main.py --use_weights --name_model debiased`, and get
> Unnamed: 0 acc_dev auc_dev acc_test auc_test acc_eec auc_eec efpr_eec efnr_eec eauc_eec
> 0 0 0.948390 0.926629 0.946831 0.924152 0.945013 0.994845 1.530650 0.312191 0.735914
> 1 1 0.947032 0.926879 0.946942 0.924213 0.942767 0.995171 1.368950 0.714149 0.935193
> 2 2 0.946942 0.928191 0.946257 0.925692 0.907359 0.980726 1.520012 3.543188 2.167654
> 3 3 0.947247 0.927384 0.946437 0.924081 0.975328 0.998819 0.995368 0.238841 0.521595
> 4 4 0.947143 0.926533 0.946298 0.923519 0.944190 0.993905 1.421833 1.408007 1.140447
> 5 5 0.946755 0.925790 0.946319 0.923033 0.946411 0.991710 0.733161 2.226244 1.317227
> 6 6 0.947960 0.926707 0.947012 0.923332 0.930228 0.984109 0.884181 4.361955 2.362223
> 7 7 0.946845 0.926851 0.945903 0.924031 0.954757 0.997575 1.276729 0.296264 0.736440
> 8 8 0.946104 0.926993 0.945806 0.924173 0.954443 0.997020 1.011461 0.324356 0.546204
> 9 9 0.946935 0.927221 0.945993 0.923317 0.971279 0.997793 1.233777 0.579832 0.770537
> 
> The mean for EFPR and EFNR for the baseline and instance weighting is pretty similar, with EFPR and EFNR for the baseline model being slightly lower than the model with instance weighting.
> 
> Is there something I'm missing, or some hyperparameter set I should be tuning for?
> Thank you!

Hi, @pruksmhc . It's a little weird. The two results you reported are identical. I think there may be some problems here.

**Comment by ghzhang233** [comment]
Hi, @pruksmhc . Just in case you forget, I would like to note that, if you run two experiments in the same folder, you need to specify the _--name_model_ parameter. Otherwise, the results will mess up.

**Comment by pruksmhc** [comment]
Ah, I think I might have copy-pasted the same thing twice in the original one. 
When I put the seed to be 0, those experiments I specified the name_model parameter and got the below results. 

acc_dev auc_dev acc_test auc_test acc_eec auc_eec efpr_eec efnr_eec eauc_eec
count 10.000000 10.000000 10.000000 10.000000 10.000000 10.000000 10.000000 10.000000 10.000000
mean 0.941833 0.919084 0.942738 0.921996 0.943946 0.991490 0.946352 1.628285 1.072813
std 0.003356 0.001476 0.003694 0.001323 0.019085 0.005198 0.399517 0.914916 0.411891
min 0.936298 0.916860 0.936201 0.919865 0.917442 0.982796 0.427023 0.423484 0.616845
25% 0.939691 0.918310 0.941114 0.921200 0.932850 0.987383 0.829629 0.896174 0.747794
50% 0.943538 0.918872 0.944345 0.922198 0.939613 0.992211 0.900189 1.588936 1.022235
75% 0.944227 0.920286 0.945635 0.922841 0.958763 0.995717 0.933675 2.255161 1.276882
max 0.944989 0.921355 0.945793 0.923734 0.973891 0.997633 1.708225 3.302407 1.814090

With debiasing, numbers seem to get worse:
acc_dev auc_dev acc_test auc_test acc_eec auc_eec efpr_eec efnr_eec eauc_eec
count 10.000000 10.000000 10.000000 10.000000 10.000000 10.000000 10.000000 10.000000 10.000000
mean 0.943308 0.918551 0.944318 0.921334 0.926287 0.987116 1.165896 2.200072 1.423375
std 0.000776 0.001829 0.000867 0.001990 0.012230 0.005744 0.425408 1.236855 0.399914
min 0.941907 0.915953 0.942524 0.917727 0.911551 0.973537 0.212848 0.436604 0.925831
25% 0.942884 0.917561 0.943938 0.920590 0.918686 0.986615 0.977410 1.535752 1.177314
50% 0.943434 0.918699 0.944373 0.921697 0.923495 0.987332 1.194725 1.884363 1.337604
75% 0.943630 0.918997 0.944830 0.922080 0.929108 0.989980 1.509363 3.033730 1.697343
max 0.944795 0.921647 0.945751 0.924552 0.951165 0.995651 1.631650 4.361005 2.192520

**Comment by pruksmhc** [comment]
I can't seem to reopen this issue, but tagging you just for exposure: @ghzhang233

**Comment by ghzhang233** [comment]
Hi, @pruksmhc . This is very weird. I may need a few days to check the codes. I will let you know when I am done.

**Comment by pruksmhc** [comment]
Thanks!

**Comment by ghzhang233** [comment]
Hi, @pruksmhc , I just re-ran the codes and got the following results, which are consistent with those reported in the paper. There are some numerical fluctuations due to the Inter-Example Correlations of IPTTS, as stated in https://arxiv.org/abs/2004.13606. I just updated the codes a little bit by modifying the dataset split method, but I am not sure if this is the cause for the problem you reported. Could you please also share what data are you using?
![image](https://user-images.githubusercontent.com/15927725/98458421-7b8f5200-21cb-11eb-9a2a-df5b8ad541da.png)

**Comment by pruksmhc** [comment]
I was using the data from the README in Jigsaw-Toxicity (train.csv), and in the code, it seems that it randomly splits the train set into train, val, and test. I am rerunning with this specified test split indices and will let you know if i'm able to replicate.
